packages/core/src/components/graphs/bar-stacked.ts

```diff
@@ -292,6 +294,27 @@ export class StackedBar extends Bar {
 			});
 	}
 
+	protected getBarWidth() {
+		const options = this.getOptions();
+		if (Tools.getProperty(options, "bars", "width")) {
+			return options.bars.width;
+		}
+		const mainXScale = this.services.cartesianScales.getMainXScale();
+		const chartWidth = DOMUtils.getSVGElementSize(this.parent, {
+			useAttrs: true,
+		}).width;
+	
+		const numberOfDomainValues = this.model.getStackKeys().length;
+	
+		if (!mainXScale.step) {
+			return Math.min(
+				options.bars.maxWidth,
+				(chartWidth * 0.25) / numberOfDomainValues
+			);
+		}
+		return Math.min(options.bars.maxWidth, mainXScale.step() / 2);
+	}
+	
```

Comment:
this seems to exist inside `bar.ts`
https://github.com/carbon-design-system/carbon-charts/blob/c600eafe6996eb1678f56850abdfbb907edb75d5/packages/core/src/components/graphs/bar.ts#L7-L27

We probably need this bit removed and just need the addition to `model.ts`

Reply:
Hi @theiliad, thanks for feedback. I have looked more closely into the `getStackKeys()` function.

https://github.com/carbon-design-system/carbon-charts/blob/fe4d592991e9cbf68c4b9efab0be4aa46dfcd416/packages/core/src/model/model.ts#L302-L344

As we see it bases on `getDisplayData()` function and just filters out the duplicates and sorts the data effectively giving us the number of bars, and that's the data we want in general in this case, not the number of data entries.

Since we use in `bar.ts` `getDisplayData()` data count, the use of `getStackKeys()` by default in `bar.ts` will only cause to filter out the duplicates, thus not impacting simple bar chart and fixing the issue in stacked bar charts.

We do not have a redundancy this way and we are just passing right data (so the number of bars instead of number of data entries) to determine the bar width